plugins/inputs/redis/redis.go

```diff
@@ -321,6 +328,48 @@ func gatherKeyspaceLine(
 	}
 }
 
+// Parse the special Keyspace line at end of redis stats
+// This is a special line that looks something like:
+//     db0:keys=2,expires=0,avg_ttl=0
+// And there is one for each db on the redis instance
+//
+// TODO: write code comments
+func gatherCommandstatsLine(
+	name string,
+	line string,
+	acc telegraf.Accumulator,
+	global_tags map[string]string,
+) {
+	if !strings.Contains(line, "calls=") {
+		return
+	}
+
+	fields := make(map[string]interface{})
+	tags := make(map[string]string)
+	for k, v := range global_tags {
+		tags[k] = v
+	}
+	tags["cmdstat"] = strings.TrimPrefix(name, "cmdstat_")
+	stats := strings.Split(line, ",")
+	for _, stat := range stats {
+		kv := strings.Split(stat, "=")
+		var (
+			ival interface{}
+			err  error
+		)
+		// only calls that always be int
+		if kv[0] == "calls" {
+			ival, err = strconv.ParseInt(kv[1], 10, 64)
+		} else {
+			ival, err = strconv.ParseFloat(kv[1], 64)
+		}
+		if err == nil {
+			fields[kv[0]] = ival
+		}
+	}
+	acc.AddFields("redis_commandstats", fields, tags)
```

Comment:
What if `fields` is empty?

Reply:
It’s hard to be empty because redis are not returns any line if it has no any command calls. Except that commandstats return the result incorrectly.